wgsl/index.bs

```diff
@@ -897,22 +897,65 @@ A <dfn noexport>type rule</dfn> has two parts:
     * How the expression is used in a statement.
     * Optionally, other static context.
 
-Each distinct type parameterization for a type rule is called an <dfn noexport>overload</dfn>.
-For example, [[#arithmetic-expr|unary negation]] (an expression of the form `-`|e|)
-has eight overloads, because its type rules are parameterized by a type |T| that can be any of:
-* [=i32=]
-* vec2&lt;i32&gt;
-* vec3&lt;i32&gt;
-* vec4&lt;i32&gt;
-* [=f32=]
-* vec2&lt;f32&gt;
-* vec3&lt;f32&gt;
-* vec4&lt;f32&gt;
+Type rules may have <dfn noexport>type parameters</dfn> in their preconditions and conclusions.
+When a type rule's conclusion or preconditions contain type parameters,
+we say it is <dfn noexport>parameterized</dfn>.
+When they do not, we say the rule is <dfn noexport>concrete</dfn>.
+We can make a concrete type rule from a parameterized one
+by substituting a type for each of its type parameters,
+using the same type for all occurrences of a given parameter in the rule.
+An assignment of types to a rule's type parameters
+is called a <dfn noexport>substitution</dfn>.
+
+For example, the two type rules for [[#arithmetic-expr|unary negation]] (an expression of the form `-`|e|)
```

Comment:
On the other hand, even in the new form, it's a single rule with a parameterization T that ranges over i32 and f32 and vectors of it. So it can still work with minor reworking to make cardinalities match the text.

     For example ,the type rule for unary negation (an expression of the form `-`|e|) is as follows:

...

      This is a parameterized rule, where the parameter T ....


The key is that you make substitutions for T, not that there are two original rules.